app/src/main/java/com/lagradost/cloudstream3/ui/download/DownloadFragment.kt

```diff
@@ -247,4 +276,33 @@ class DownloadFragment : Fragment() {
             binding?.downloadStreamButton?.extend()
         }
     }
+
+    // Open local video from files using content provider x safeFile
+    private val videoResultLauncher = registerForActivityResult(
+        ActivityResultContracts.StartActivityForResult()
+    ) { result ->
+        if (result.resultCode == Activity.RESULT_OK) {
+            val intentData = result?.data?.data
+            intentData.let { selectedVideoUri ->
+                val name = SafeFile.fromUri(
+                    context ?: return@let ,
+                    selectedVideoUri ?: return@registerForActivityResult
+                )?.name()
+
+                activity?.navigate(
```

Comment:
Use this instead
https://github.com/recloudstream/cloudstream/blob/55a0eb66cb5fa3833fd3a84f6cd231ece7e7edeb/app/src/main/java/com/lagradost/cloudstream3/ui/player/DownloadedPlayerActivity.kt#L50